docs/data_sources.md

```diff
@@ -112,6 +112,22 @@ Full parameters:
 directive @stub(value: JSON!)
 ```
 
+## @mergeWithStub
+
+`@mergeWithStub` works similarly to `@stub`, but merges the stub result with the existing resolver for the field.
+
+```graphql
+type User {
+    firstName: String!
+    lastName: String!
+    fullName: String // not supplied by the resolver
+}
+
+type Query {
+    getUser: User! @mergeWithStub(value: "{{ fullName: `${source.firstName} ${source.lastName}` }}")
```

Comment:
Kind of a nitpick, but probably better to promote using the `JSON` scalar here instead of a string
```suggestion
    getUser: User! @mergeWithStub(value: {fullName: "${source.firstName} ${source.lastName}"})
```

Reply:
I kept the injection implementation the same as stub, so this won't work on either of them (they only inject for string, anything else is returned as is)

Since I don't really see a reason not to, I'll just change both to also support deep injection (in addition to making this docs change)